### PR TITLE
Linking

### PR DESCRIPTION
**Pull request by adrian-lara**
WARNING: This PR includes a DB migration that automatically runs a match-merge-link round for all records across all cycles across all organizations. (Done to establish a baseline of consistent data)

#### Any background context you want to provide?
In order to support year-over-year analysis within SEED, it's necessary to establish `links` between records across cycles. This requires certain record consistency conditions be met, and logic has been included to allow SEED to continually meet these conditions.

Specifically, if two records are linked across cycles, they must match. Implications of this include unlinking records if they no longer match.

For more information on matching, merging, and linking, see the `docs/source/matching.rst` file.

#### What's this PR do?
Quick side note - inventory detail page highlights columns with values changes amongst master and historical was added into this PR.

This PR implements linking between records across Cycles using the canonical Property and TaxLot models (which are associated via the corresponding -Views).

Backend:
- Establishes a baseline of consistent data via DB migration (see warning above).
- Incorporates a linking round whenever there was previously a match-merge round, effectively replacing the match-merge round with the notion of a match-merge-linking round.  This happens in the following cases:
  - post record edit
  - post records merge
  - during the import of incoming records (only affects matches amongst existing records)
- Removes links whenever records are unmerged.
- Removes links whenever records no longer match. This happens in the following cases:
  - post unmerge of records (neither are linked and a match-merge-round isn't run on them right away)
  - post edit of a record
  - user changes matching criteria for an organization. This can be an intensive calculation that can create a state of the database that would be difficult to reverse. Given that, a preview has been created that will show users what their database would look like before confirming the changes.

Frontend UI/UX:
- Links for an individual record can be viewed via Cross-Cycles detail page/tab.
- Links can be viewed in aggregate via a new Cross-Cycles list page/tab.
- As mentioned in the above section, when a user changes matching criteria, they are able to view a preview of the changed data before confirming. The speed of the preview generation and actual calculations depend heavily on the number of record merges needed.
- Notifications are provided when merges are executed and links are created during match-merge-linking rounds.

Also, reprioritization of automatically merged records is included in this PR. The -State  `updated` fields are now used to prioritize these automatic merges. See https://github.com/SEED-platform/seed/pull/1957/commits/f99827df32d308abe8426ebd6461398003f45c94

#### How should this be manually tested?
- Generally, testing should involve going through all the match-merge-linking round scenarios mentioned above and verify links are established (or broken) via the two Cross-Cycle pages. 

#### What are the relevant tickets?
#1946 
#1232 
#1591

### CODE REVIEW

File: seed/data_importer/match.py [hunk 1]
```diff
@@ -364,6 +376,30 @@ def states_to_views(unmatched_state_ids, org, cycle, StateClass):
     return list(set(processed_views)), duplicate_count, new_count + matched_count
 
 
+def link_views(merged_views, ViewClass):
```
Comment (adrian-lara) [inline seed/data_importer/match.py]: This is the most notable change in this file. The other changes are mostly from a refactor of matching criteria getters and variable/function renaming.

File: seed/static/seed/js/controllers/confirm_column_settings_modal_controller.js [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        warnings_acknowledged: true,
+      }
+
+      // Check if PM Property ID is actually being removed from matching criteria
+      if (_.find($scope.change_summary_data, {column_name: "pm_property_id", is_matching_criteria: false})) {
+        $scope.checks.matching_criteria_excludes_pm_property_id = true;
+        $scope.checks.warnings_acknowledged = false;
+      }
+
+      // Preview
```
Comment (adrian-lara) [inline seed/static/seed/js/controllers/confirm_column_settings_modal_controller.js]: There are some duplicate lines of code between this and the cycles list controller but there are definitely some necessary and nuanced differences. It's likely possible to DRY this up a bit by abstracting the two but I didn't place a high priority on it as it's maybe not the most cost efficient talk. I'm happy to revisit this later though

**Comment by RDmitchell** [comment]
Older Issues that should be checked: 
#6

File: config/settings/common.py [hunk 3]
```diff
@@ -207,7 +207,7 @@
          CeleryDatetimeSerializer.seed_loads,
          content_type='application/json', content_encoding='utf-8')
 CELERY_WORKER_MAX_TASKS_PER_CHILD = 1
-CELERY_ACCEPT_CONTENT = ['seed_json']
+CELERY_ACCEPT_CONTENT = ['seed_json', 'pickle']
```
[the remark this answers is not in the document]
Comment (nllong) [reply config/settings/common.py]: 👍

**Comment by RDmitchell** [comment]
@adrian-lara / @nllong -- the statement in the description about the db migration seems a bit "scary" to me, depending on how users have mapped their data in the past. I guess we will have to test this on real user data, but I am wondering if doing a forced migration is the best strategy. 

I might want to have a session with you all to have you go over the implications of this.

**Pull request merged**